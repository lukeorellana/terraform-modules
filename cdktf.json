{
  "language": "typescript",
  "app": "npx ts-node",
  "terraformProviders": [],
  "projectId": "1b5f16bd-67ac-4af6-a33f-7bcafff141eb"
}