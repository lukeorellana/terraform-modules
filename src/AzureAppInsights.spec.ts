import { Testing, TerraformStack} from 'cdktf';
import {AzurermProvider} from "@cdktf/provider-azurerm";
import { AzureApplicationInsights } from './AzureApplicationInsights';
import 'cdktf/lib/testing/adapters/jest';


describe('AzureApplicationInsights-Snapshot', () => {
  it('renders a AzureApplicationInsights without tags', () => {

    const synthed = Testing.synthScope((stack) => {
      new AzureApplicationInsights(stack, 'testAppi', {
        name: 'appi-test',
        location: 'eastus',
        resource_group_name: "rg-test",
        application_type: "web",
        daily_data_cap_in_gb: 10,
        daily_data_cap_notification_disabled: true,

      });
    });
  
    expect(synthed).toMatchSnapshot();
  });

  it('renders a AzureApplicationInsights with tags', () => {
    const synthed = Testing.synthScope((stack) => {
      new AzureApplicationInsights(stack, 'testAppi', {
        name: 'appi-test',
        location: 'eastus',
        resource_group_name: "rg-test",
        application_type: "web",
        daily_data_cap_in_gb: 10,
        daily_data_cap_notification_disabled: true,
        tags: {
          environment: "test",
        },

      });
    });
    expect(synthed).toMatchSnapshot();
  });
});


describe("AzureApplicationInsights-Terraform", () => {
  it("check if the produced terraform configuration is valid", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");
    
    new AzurermProvider(stack, "azureFeature", {
      features: {},
    });


    new AzureApplicationInsights(stack, 'testAppi', {
      name: 'appi-test',
      location: 'eastus',
      resource_group_name: "rg-test",
      application_type: "web",
      daily_data_cap_in_gb: 10,
      daily_data_cap_notification_disabled: true,
      tags: {
        environment: "test",
      },

    });
    

    // We need to do a full synth to validate the terraform configuration
    expect(Testing.fullSynth(stack)).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", {
      features: {},
    });

    new AzureApplicationInsights(stack, 'testAppi', {
      name: 'appi-test',
      location: 'eastus',
      resource_group_name: "rg-test",
      application_type: "web",
      daily_data_cap_in_gb: 10,
      daily_data_cap_notification_disabled: true,
      tags: {
        environment: "test",
      },

    });


    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(stack)).toPlanSuccessfully();
  });
});
