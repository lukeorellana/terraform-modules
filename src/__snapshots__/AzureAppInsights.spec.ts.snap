// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzureApplicationInsights-Snapshot renders a AzureApplicationInsights with tags 1`] = `
"{
  "output": {
    "app_id": {
      "value": "\${azurerm_application_insights.testAppi_appinsights_06357BD3.app_id}"
    },
    "connection_string": {
      "sensitive": true,
      "value": "\${azurerm_application_insights.testAppi_appinsights_06357BD3.connection_string}"
    },
    "id": {
      "value": "\${azurerm_application_insights.testAppi_appinsights_06357BD3.id}"
    },
    "instrumentation_key": {
      "sensitive": true,
      "value": "\${azurerm_application_insights.testAppi_appinsights_06357BD3.instrumentation_key}"
    },
    "name": {
      "value": "\${azurerm_application_insights.testAppi_appinsights_06357BD3.name}"
    }
  },
  "resource": {
    "azurerm_application_insights": {
      "testAppi_appinsights_06357BD3": {
        "application_type": "web",
        "daily_data_cap_in_gb": 10,
        "daily_data_cap_notifications_disabled": true,
        "location": "eastus",
        "name": "appi-test",
        "resource_group_name": "rg-test",
        "tags": {
          "environment": "test"
        }
      }
    }
  }
}"
`;

exports[`AzureApplicationInsights-Snapshot renders a AzureApplicationInsights without tags 1`] = `
"{
  "output": {
    "app_id": {
      "value": "\${azurerm_application_insights.testAppi_appinsights_06357BD3.app_id}"
    },
    "connection_string": {
      "sensitive": true,
      "value": "\${azurerm_application_insights.testAppi_appinsights_06357BD3.connection_string}"
    },
    "id": {
      "value": "\${azurerm_application_insights.testAppi_appinsights_06357BD3.id}"
    },
    "instrumentation_key": {
      "sensitive": true,
      "value": "\${azurerm_application_insights.testAppi_appinsights_06357BD3.instrumentation_key}"
    },
    "name": {
      "value": "\${azurerm_application_insights.testAppi_appinsights_06357BD3.name}"
    }
  },
  "resource": {
    "azurerm_application_insights": {
      "testAppi_appinsights_06357BD3": {
        "application_type": "web",
        "daily_data_cap_in_gb": 10,
        "daily_data_cap_notifications_disabled": true,
        "location": "eastus",
        "name": "appi-test",
        "resource_group_name": "rg-test"
      }
    }
  }
}"
`;
